src/mesh/topology.rs

```diff
@@ -19,43 +12,74 @@ use super::{Face, Mesh};
 /// contain before it spills into heap. Implementation detail.
 const MAX_INLINE_NEIGHBOR_COUNT: usize = 8;
 
+pub type VertexToFaceRelation = SmallVec<[u32; MAX_INLINE_NEIGHBOR_COUNT]>;
+
+/// Calculates topological relations (neighborhood) of mesh vertex -> faces.
+/// Returns a Map (key: vertex index, value: list of its neighboring faces indices)
+pub fn calculate_vertex_to_face_topology(mesh: &Mesh) -> Vec<VertexToFaceRelation> {
+    calculate_vertex_to_face_topology_from_components(mesh.faces(), cast_u32(mesh.vertices().len()))
+}
+
+pub fn calculate_vertex_to_face_topology_from_components(
```

Comment:
Hmm, since this split is trivial to make, let's make it once we need it. It's not used anywhere else in this PR, so I would prefer if we only had just one function - the one that takes the whole mesh. Do you have some concrete plans with this?

Reply:
I already have another PR ready using this.